### PR TITLE
Do not resolve addresses in remote connection info

### LINKED ISSUE #35658: Remote cluster info returns fatally for invalid hostnames
I added a cluster with an invalid hostname:

```
PUT _cluster/settings
{
  "persistent": {
    "cluster": {
      "remote": {
        "jentest": {
          "seeds": [
            "testname:1234"
          ]
        }
      }
    }
  }
}
```

this is causing `GET _remote/info` to return fatally

```
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "unknown host [testname]"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "unknown host [testname]",
    "caused_by": {
      "type": "unknown_host_exception",
      "reason": "testname: nodename nor servname provided, or not known"
    }
  },
  "status": 400
}
```

For our Remote Clusters management UI feature, we're using `_remote/info` to fetch the list and info of clusters for display. Ideally it would still return the list of all clusters, maybe with the error attached to the corresponding cluster.

### LINKED ISSUE #36667: Removing remote cluster that is in use by auto-follow pattern causes crash
I have a remote cluster which is in use by a CCR auto-follow pattern:

`GET _remote/info`:
```
{
  "prod2" : {
    "seeds" : [
      "127.0.0.1:9400"
    ],
    "connected" : true,
    "num_nodes_connected" : 1,
    "max_connections_per_cluster" : 3,
    "initial_connect_timeout" : "30s",
    "skip_unavailable" : false
  }
}
```

`GET _ccr/auto_follow`:
```
{
  "patterns" : [
    {
      "name" : "pattern1",
      "pattern" : {
        "remote_cluster" : "prod2",
        "leader_index_patterns" : [
          "test*"
        ],
        "follow_index_pattern" : "{{leader_index}}"
      }
    }
  ]
}
```

When I remove the remote cluster, ES throws repeated errors until it crashes with `ERROR ES exited with code 126`:

```
PUT _cluster/settings
{
  "persistent": {
    "cluster": {
      "remote": {
        "prod2": {
          "seeds" : null,
          "skip_unavailable": null
        }
      }
    }
  }
}
```

```
   │ info [o.e.x.c.a.AutoFollowCoordinator] [Jens-MacBook-Pro.local] failure occurred while fetching cluster state for auto follow pattern [pattern1]
   │      java.lang.IllegalArgumentException: unknown cluster alias [prod2]
   │      	at org.elasticsearch.transport.RemoteClusterService.getRemoteClusterClient(RemoteClusterService.java:495) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.client.node.NodeClient.getRemoteClusterClient(NodeClient.java:123) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch
   │ info .xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda$getRemoteClusterState$0(AutoFollowCoordinator.java:165) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.CcrLicenseChecker.checkRemoteClusterLicenseAndFetchClusterState(CcrLicenseChecker.java:178) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.getRemoteClusterState(AutoFollowCoordinator.java:161) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.start(AutoFollowCoordinator.java:275) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.finalise(AutoFollowCoordinator.java:407) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$AutoFollower.lambda$start$1(AutoFollowCoordinator.java:289) [x-pack-ccr-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
   │      	at org.elasticsearch.xpack.ccr.action.AutoFollowCoordinator$2.lambda
```

### PR DESCRIPTION
**Pull request by jasontedor**
The remote connection info API leads to resolving addresses of seed nodes when invoked. This is problematic because if a hostname fails to resolve, we would not display any remote connection info. Yet, a hostname not resolving can happen across remote clusters, especially in the modern world of cloud services with dynamically chaning IPs. Instead, the remote connection info API should be providing the configured seed nodes. This commit changes the remote connection info to display the configured seed nodes, avoiding a hostname resolution. Note that care was taken to preserve backwards compatibility with previous versions that expect the remote connection info to serialize a transport address instead of a string representing the hostname.

Closes #35658

### CODE REVIEW

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests 2

File: server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java [hunk 1]
```diff
@@ -57,7 +63,17 @@
     }
 
     public RemoteConnectionInfo(StreamInput input) throws IOException {
-        seedNodes = input.readList(TransportAddress::new);
+        if (input.getVersion().onOrAfter(Version.V_7_0_0)) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java]: Is the bwc aspect of serialization sufficiently tested? I do see `RemoteClusterConnectionTests#testRemoteConnectionInfoBwComp`, just wondering whether that is good enough.

**Review by martijnvg — APPROVED** [review]
👍 - I left one testing related question

File: server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java (hunk 1 above)
Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java]: I think that's good enough. It shows that if a < 7.0.0 node sends us how it serializes messages today that we can de-serialize it and get what we expect, and shows that if we are sending to a < 7.0.0 that we serialize as that node expects. Also, those versions will change to 6.6.0 when this is backported.

**Comment by jasontedor** [comment]
Thanks for reviewing @martijnvg.

**Pull request merged**